### PR TITLE
Bump typescript plugin version

### LINKED ISSUE #154: bundle/reload before TS compile
as title
as title

repo: https://github.com/valen214/ts-lag-one-cycle

modification after fork:
ran `node scripts/setupTypeScript.js`
added `import { getText } from "./Foo"; console.log(getText());` to `App.svelte`
added `Foo.ts`
nothing else

### PR DESCRIPTION
**Pull request by dummdidumm**
Fixes reload problems. Fixes #154

Although #154 was closed the fix for it never made it into the template.

### CODE REVIEW

**Comment by benmccann** [comment]
I expect there are still more reload issues. `rollup-plugin-typescript` [overrides what's in `tsconfig.json` for three option values](https://github.com/rollup/plugins/blob/0e747cd609dfc851a2d7b330400616d5e068c78f/packages/typescript/src/options/interfaces.ts#L9) and one of those overrides makes the plugin crash when you have a syntax error. I found that you need to override the plugin's override of the value from `tsconfig.json` back to the original value in the rollup config.

We have three options:
* Send a patch to the plugin remove the overrides, which I think are pretty unexpected. I think this would be nice to do as a good citizen, but haven't had time to do it yet. See the upstream bug here: https://github.com/rollup/plugins/issues/272
* Switch to `rollup-plugin-typescript2`. I've only used the plugins on a few projects, but found it to be more trouble free. See https://github.com/sveltejs/template/pull/161 for a PR that would make that change
* Add the workaround of `noEmitOnError: false` in the rollup config

**Comment by dummdidumm** [comment]
Switching to `rollup-plugin-typescript2` introduced another bug for me: I cannot import TypeScript files in Svelte files if I omit the `.ts`. So `import ... from './Test.ts` works (but is errored in the IDE which is correct) but `import ... from './Test'` doesn't.

I'll try to dig into the official plugin some more.

**Comment by dummdidumm** [comment]
I added a fix for the emit behavior in #173

**Review by benmccann — APPROVED** [review]

File: scripts/setupTypeScript.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const packageJSON = JSON.parse(fs.readFileSync(path.join(projectRoot, "package.j
 packageJSON.devDependencies = Object.assign(packageJSON.devDependencies, {
   "svelte-check": "^1.0.0",
   "svelte-preprocess": "^4.0.0",
-  "@rollup/plugin-typescript": "^4.0.0",
+  "@rollup/plugin-typescript": "^5.0.0",
```
Comment (benmccann) [inline scripts/setupTypeScript.js:27]: It looks like 6.0.0 is out now!

**Pull request merged**